inbm/dispatcher-agent/dispatcher/packagemanager/local_repo.py

```diff
@@ -131,8 +131,15 @@ def delete_all(self) -> None:
         """
         repo_path = self.get_repo_path()
         logger.debug("Erasing files directly underneath " + repo_path)
-        for the_file in os.listdir(repo_path):
-            file_path = os.path.join(self.get_repo_path(), the_file)
-            if os.path.isfile(file_path):
-                logger.debug("... unlinking: " + file_path)
-                os.unlink(file_path)
+        if not os.path.exists(repo_path):
+            raise DispatcherException(f"cannot delete files from {repo_path}: does not exist")
+        if not os.path.isdir(repo_path):
```

Comment:
I wonder about putting the path in the error message.  I know there is an SDL item about not providing information that a hacker could use.  If it's different and provided by the user, then I think it's alright.  Otherwise, we might want to remove that information.

Reply:
@nmgaston good point.. this could appear in an error message. Maybe I can make the exception more generic and do a logger.debug on the repo_path

Reply:
@gblewis1 - that would be good.